src/rendering/renderers/shared/instructions/InstructionSet.ts

```diff
@@ -2,32 +2,47 @@ import { uid } from '../../../../utils/data/uid';
 
 import type { Instruction } from './Instruction';
 
+/**
+ * A set of instructions that can be executed by the renderer.
+ * Basically wraps an array, but with some extra properties that help the renderer
+ * to keep things nice and optimised.
+ *
+ * Note:
+ * InstructionSet.instructions contains all the instructions, but does not resize (for performance).
+ * So for the true length of the instructions you need to use InstructionSet.instructionSize
+ * @memberof rendering
+ */
 export class InstructionSet
 {
-    public uid = uid('instructionSet');
-    public instructions: Instruction[] = [];
+    /** a unique id for this instruction set used through the renderer */
+    public readonly uid = uid('instructionSet');
+    /** the array of instructions */
+    public readonly instructions: Instruction[] = [];
+    /** the actual size of the array (any instructions passed this should be ignored) */
     public instructionSize = 0;
+    /** allows for access to the render pipes of the renderer */
     public renderPipes: any;
 
+    /** reset the instruction set so it can be reused set size back to 0 */
     public reset()
     {
         this.instructionSize = 0;
     }
 
+    /**
+     * Add an instruction to the set
+     * @param instruction - add an instruction to the set
+     */
     public add(instruction: Instruction)
     {
         this.instructions[this.instructionSize++] = instruction;
     }
 
+    /** Log the instructions to the console (for debugging) */
     public log()
```

Comment:
does this need to be added to the api docs?

seems like we should keep this as an internal API

Reply:
im cool with that. let me. update